ports/stm32/main.c

```diff
@@ -508,11 +508,8 @@ void stm32_main(uint32_t reset_mode) {
     mp_thread_init();
     #endif
 
-    // Stack limit should be less than real stack size, so we have a chance
-    // to recover from limit hit.  (Limit is measured in bytes.)
-    // Note: stack control relies on main thread being initialised above
-    mp_stack_set_top(&_estack);
-    mp_stack_set_limit((char *)&_estack - (char *)&_sstack - 1024);
+    // Stack limit init.
+    mp_cstack_init_with_top(&_estack, (char *)&_estack - (char *)&_sstack);
```

Comment:
I'm pretty sure you'll need to add `#define MICROPY_STACK_CHECK_MARGIN (1024)` to `mpconfigport.h` to make this work the same as before.

Reply:
Thanks, I've added that now.